cmd/jaeger/internal/extension/jaegerquery/server_test.go

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package jaegerquery
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/extension"
+	noopmetric "go.opentelemetry.io/otel/metric/noop"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/jaegerstorage"
+	queryApp "github.com/jaegertracing/jaeger/cmd/query/app"
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc"
+	"github.com/jaegertracing/jaeger/pkg/jtracer"
+	memoryCfg "github.com/jaegertracing/jaeger/pkg/memory/config"
+	"github.com/jaegertracing/jaeger/pkg/tenancy"
+	"github.com/jaegertracing/jaeger/pkg/testutils"
+	depsmocks "github.com/jaegertracing/jaeger/storage/dependencystore/mocks"
+	spanstoremocks "github.com/jaegertracing/jaeger/storage/spanstore/mocks"
+)
+
+type storageHost struct{}
+
+func (host storageHost) ReportFatalError(err error) {
+}
+
+func (host storageHost) GetExtensions() map[component.ID]component.Component {
+	return map[component.ID]component.Component{
+		component.NewID("jaeger_storage"): makeStorageExtension("jaeger_storage"),
+	}
+}
+
+func (storageHost) GetFactory(_ component.Kind, _ component.Type) component.Factory {
+	return nil
+}
+
+func (storageHost) GetExporters() map[component.DataType]map[component.ID]component.Component {
+	return nil
+}
+
+func Test_NewSever(t *testing.T) {
+	telemetrySettings := component.TelemetrySettings{
+		Logger: zap.NewNop(),
+	}
+	server := newServer(createDefaultConfig().(*Config), telemetrySettings)
+	assert.NotNil(t, server)
+	assert.NotNil(t, server.config)
+	assert.NotNil(t, server.logger)
+}
+
+func Test_Dependencies(t *testing.T) {
+	expectedDependencies := []component.ID{component.NewID(component.Type("jaeger_storage"))}
```

Comment:
move this to just before the assert

Reply:
also, is `component.NewID(component.Type("jaeger_storage"))` necessary? There is a declared constant for that